sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -798,6 +798,156 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("array_repeat function") {
```

Comment:
We don't need so many cases here. We only need to verify the api works end to end.
Evaluation checks of the function should be in `CollectionExpressionsSuite`.